packages/libs/web-common/src/controllers/LegacyMapRedirectHandler.tsx

```diff
@@ -81,7 +89,7 @@ export function LegacyMapRedirectHandler({
         const additionalAnalysisConfig = {
           ...baseAdditionalAnalysisConfig,
           ...descriptorConfig,
-        };
+        } as AdditionalAnalysisConfig;
```

Comment:
Instead of using `as`, we should declare the type of the variable:

`const additionalAnalysisConfig: AdditionalAnalysisConfig = {`

This will require declare the components that make up this value (`baseAdditionalAnalysisConfig`, and `descriptorConfig`). I think this will reveal a subtle type conflict.

Generally, we should avoid `as` if possible, and document why we are using it if we need it. This isn't something we've really enforced before.